hydra-cluster/src/Hydra/Cluster/Scenarios.hs

```diff
@@ -0,0 +1,84 @@
+{-# LANGUAGE TypeApplications #-}
+
+module Hydra.Cluster.Scenarios where
+
+import Hydra.Prelude
+
+import CardanoClient (queryTip)
+import CardanoNode (RunningNode (RunningNode))
+import Control.Lens ((^?))
+import Data.Aeson (object, (.=))
+import Data.Aeson.Lens (key, _Number)
+import qualified Data.Set as Set
+import Hydra.Cardano.Api (Lovelace, NetworkId, selectLovelace)
+import Hydra.Cluster.Faucet (Marked (Fuel), queryMarkedUTxO, seedFromFaucet)
+import Hydra.Cluster.Fixture (Actor (Alice), actorName, alice, aliceSk)
+import Hydra.Cluster.Util (chainConfigFor, keysFor)
+import Hydra.Ledger (IsTx (balance))
+import Hydra.Ledger.Cardano (Tx)
+import Hydra.Logging (Tracer, traceWith)
+import Hydra.Options (networkId, startChainFrom)
+import HydraNode (EndToEndLog (..), input, output, send, waitFor, waitMatch, withHydraNode)
+
+-- TODO: The 'RunningNode' should convey the networkId
+singlePartyHeadFullLifeCycle ::
+  Tracer IO EndToEndLog ->
+  FilePath ->
+  NetworkId ->
+  RunningNode ->
+  IO ()
+singlePartyHeadFullLifeCycle tracer workDir networkId node = do
```

Comment:
It's a bit sad there is some redundancy with what's in the `test/` directory, it feels there is some "testing DSL" waiting to be exposed here, perhaps as some module in the `src/` for hydra-cluster so that it's reused across tests and exes?